week-3/extended-project/champion-performance/http_server.rb

```diff
@@ -0,0 +1,43 @@
+require 'socket'
+require 'uri'
+require './controller/controller'
+require './http_handler'
+
+WEB_ROOT = './public'
+
+
+class HomePage < RequestHandler
+  def req
+    path = File.join(WEB_ROOT, 'index.html')
+    self.send(path)
+  end
+end
+
+class RequestFile < RequestHandler
+  def req
+    request_uri = self.request[:uri]
+    path = File.join('./', request_uri)
+    self.send(path)
+  end
+end
+
+class MemberPage < RequestHandler
+  def req
+    name = self.request[:vars][:name]
```

Comment:
How do you use "[:vars][:name]" on this line?

Reply:
We are using a hash named require in order to specify the name of the members which help us to produce the html from the erb.
http://localhost:2345/?[vars]=[name]
Because we have 5 different names in the case of vars because we build a hash that could accept more keys